internal/rpc/core/status.go

```diff
@@ -59,8 +60,15 @@ func (env *Environment) Status(ctx context.Context) (*coretypes.ResultStatus, er
 		}
 	}
 
+	applicationInfo := coretypes.ApplicationInfo{}
+	abciInfo, _ := env.ABCIInfo(ctx)
+	if abciInfo != nil {
```

Comment:
Actually, let's kill two birds with one stone here, given that `ABCIInfo` never reports nil except in case of error:

```suggestion
	var applicationInfo coretypes.ApplicationInfo
	if abciInfo, err := env.ABCIInfo(ctx); err == nil {
		applicationInfo.Version = fmt.Sprint(abciInfo.Response.AppVersion)
```

It might also be worth logging an error in that case, but I don't think it'd help much (since if this method fails, the node is probably doomed).

Reply:
Yeah I thought so, that's why i ignored the error. This works as well